src/loclists.rs

```diff
@@ -39,39 +39,83 @@ where
     }
 }
 
+/// An offset into either the `.debug_loc` section or the `.debug_loclists` section,
+/// depending on the version of the unit the offset was contained in.
```

Comment:
Should we return an `enum` of either a `DebugLocOffset` or a `DebugLocListsOffset`? I'm concerned that we are making it easier to misuse our APIs and weakening the benefits that strongly typing offsets gives.

Reply:
Nevermind -- I understand how this works better now.

Reply:
Yeah I wasn't sure which way was better.  We could make it an enum, and then you wouldn't need to pass the unit version into the other functions, but you would still need to pass the unit address size (or we could include that in the enum too).